src/components/table/selection.ts

```diff
@@ -0,0 +1,90 @@
+export interface SelectionChangeSet {
+    selected: boolean;
+    items: any[];
+    indexes: number[];
+}
+
+export class Selection {
+    private selectedItems: Set<any>;
+    private lastToggledIndex: number;
+
+    constructor(private getDataByIndex: (number) => any) {
+        this.clear();
+    }
+
+    get size(): number {
+        return this.selectedItems.size;
+    }
+
+    get items(): any[] {
+        return Array.from(this.selectedItems);
+    }
+
+    set items(items: any[]) {
+        this.selectedItems = new Set(items);
+        this.lastToggledIndex = -1;
+    }
+
+    public has(data: any) {
+        return this.selectedItems.has(data);
+    }
+
+    public toggleSelection(index: number): SelectionChangeSet {
+        return this.toggleRange(index, index);
+    }
+
+    public toggleSelectionFromLastIndex(index: number): SelectionChangeSet {
```

Comment:
I don't know the context here (I just saw the comments because I'm subscribed to the whole repo), but you might also want to consider decorating these with `@Method()` too: https://stenciljs.com/docs/methods#method-decorator

Reply:
But this is not a component so I won't add `@Method()` to these. :)

Reply:
# 🤦‍♂️😆✅